R/fit_devmodels.R

```diff
@@ -2,98 +2,122 @@
-                              model_name == "ratkowsky" ~ "ratkowsky(temp, tmin, tmax, a, b)",
-                              model_name == "rezende" ~ "rezende(temp, q10, a, b, c)",
-                              model_name == "ssi" ~ "ssi(temp, r_tref, e, el, tl, eh, th, tref = 20)",
-                              model_name == "mod_weibull" ~ "mod_weibull(temp, a, topt, b, c)"
-                              )
-          )
+      ## I think these formulas below should be defined out of here, e.g. when defining dev_model_table
+      mutate(formula = case_when(model_name == "briere1" ~  "briere1(temp, tmin, tmax, a)",
+                                 model_name == "lactin1" ~ "lactin1(temp, a, tmax, delta_t)",
+                                 model_name == "janisch" ~ "janisch(temp, topt, dmin, a, b)",
+                                 model_name == "linear_campbell" ~ "linear_campbell(temp,intercept, slope)",
+                                 model_name == "wang" ~ "wang(temp, k, r, topt, tmin, tmax, a)",
+                                 model_name == "mod_polynomial" ~ "mod_polynomial(temp, a_0, a_1, a_2, a_3, a_4)",
+                                 model_name == "briere2" ~ "briere2(temp, tmin, tmax, a, b)",
+                                 model_name == "mod_gaussian" ~ "mod_gaussian(temp, rmax, topt, a)",
+                                 model_name == "lactin2" ~ "lactin2(temp, a, tmax, delta_t, b)",
+                                 model_name == "ratkowsky" ~ "ratkowsky(temp, tmin, tmax, a, b)",
+                                 model_name == "rezende" ~ "rezende(temp, q10, a, b, c)",
+                                 model_name == "ssi" ~ "ssi(temp, r_tref, e, el, tl, eh, th, tref = 20)",
+                                 model_name == "mod_weibull" ~ "mod_weibull(temp, a, topt, b, c)"
+      )
+      )
+
     devdata <- tibble(temp = temp,
                       dev_rate = dev_rate)
+
     ## then fit model with nlme::gnls function
-    fit_gnls <- suppressWarnings(gnls(model = reformulate(response = "dev_rate",
-                                                          termlabels = unique(start_formula_tbl$formula)),
-                                      data = devdata,
-                                      start = replace_na(start_vals, 0), #to avoid error if start values compute a NA, probably not converging
-                                      na.action = na.exclude, #to avoid problems in the model
-                                      weights = varExp(form = ~temp), #usually development at higher temperatures has higher variability due to higher mortality
-                                      control = gnlsControl(maxIter = 100,
-                                      nlsTol = 1e-07,
-                                      returnObject = TRUE)
-                                      )
-                                 )
+    #### I don't think suppressWarnings is a sensible thing to do here?
```

Comment:
The aim was to avoid default warnings indicating convergence issues, which are difficult to understand, but just removing them might not be a good option. It is not a problem to print them by default without `suppressWarnings()` since they are just a few.

Reply:
If there are convergence problems I think the user should be aware of them. Otherwise they might go on ignoring the problems. 
What we can do is to make it as easy as possible for them to interpret and fix such problems